### ISSUE TITLE
Add support for WAI-ARIA aria-live property for portions of the page updated via <f:ajax>

### ISSUE THREAD

**Issue by glassfishrobot** [javaee/javaserverfaces-spec#1159]
Currently, JSF doesn't have any explicit support for WAI-ARIA ([http://www.w3.org/WAI/intro/aria.php](http://www.w3.org/WAI/intro/aria.php)), except for the role attribute. <f:ajax> should be updated to use the aria-live attribute at the very least.

See EG thread here: [http://java.net/projects/javaserverfaces-spec-public/lists/jsr344-experts/archive/2013-01/message/5](http://java.net/projects/javaserverfaces-spec-public/lists/jsr344-experts/archive/2013-01/message/5)

**Comment by glassfishrobot** [comment]
Reported by kito75

**Comment by glassfishrobot** [comment]
kito75 said:
After looking at this in more detail, there are several attributes that one may need to apply to the element that's being updated by an Ajax:

*   role
*   aria-live
*   aria-busy
*   aria-atomic
*   aria-relevant

(See [http://www.w3.org/TR/wai-aria-practices/#LiveRegions](http://www.w3.org/TR/wai-aria-practices/#LiveRegions))

Only the page author (and possibly the component being re-rendered) know which attributes should be set, and what the values should be. Since the regions are denoted with these attributes up-front, perhaps there's not much for <f:ajax> to do after all, because the page author can either use pass-through attributes or HTML5-friendly markup to specify these attributes on the component that needs to be updated.

**Comment by glassfishrobot** [comment]
@edburns said:
I like this idea, targeting for 2.3.

**Comment by glassfishrobot** [comment]
@edburns said:
Set priority to baseline ahead of JSF 2.3 triage. Priorities will be assigned accurately after this exercise.

**Comment by glassfishrobot** [comment]
@manfredriem said:
Setting priority to Minor

**Comment by glassfishrobot** [comment]
This issue was imported from java.net JIRA JAVASERVERFACES_SPEC_PUBLIC-1159

**Comment by glassfishrobot** [comment]
**Closing this as this issue is migrated to** https://github.com/eclipse-ee4j/faces-api/issues/1159
